src/main/java/de/srendi/advancedperipherals/common/addons/refinedstorage/RSApi.java

```diff
@@ -0,0 +1,694 @@
+        }
+        AdvancedPeripherals.debug("Could not create table from unknown resource " + resource.getClass() + " - Report this to the maintainer of ap", Level.WARN);
+        return Collections.emptyMap();
+    }
+
+    /**
+     * Helper method to get a parsed lua resourceAmount key. Currently only suppors item and fluid resources
+     *
+     * @param resourceAmount the resourceAmount
+     * @return the parsed key to a lua properties map
+     */
+    public static Map<String, Object> getObjectFromResourceAmount(@NotNull ResourceAmount resourceAmount) {
+        if (resourceAmount.resource() instanceof ItemResource) {
+            return getObjectFromItemResource(resourceAmount);
+        }
+        if (resourceAmount.resource() instanceof FluidResource) {
+            return getObjectFromFluidResource(resourceAmount);
+        }
+        AdvancedPeripherals.debug("Could not create table from unknown resourceAmount " + resourceAmount.getClass() + " - Report this to the maintainer of ap", Level.WARN);
+        return Collections.emptyMap();
+    }
+
+    public static Object parseDiskDrive(StorageNetworkNode diskDrive, InWorldNetworkNodeContainer nodeContainer) {
+        Map<String, Object> properties = new HashMap<>();
+
+        StorageConfiguration storageConfiguration = diskDrive.getStorageConfiguration();
+        CompositeStorage storage = (CompositeStorage) diskDrive.getStorage();
+
+        Set<Object> disks = new HashSet<>();
+        for (Storage disk : storage.getSources()) {
+            if (!(disk instanceof StateTrackedStorage stateTrackedStorage))
+                continue;
+
+            disks.add(parseStorageDisk(stateTrackedStorage));
+        }
+
+        properties.put("used", diskDrive.getStored());
+        properties.put("total", diskDrive.getCapacity());
+        properties.put("disks", disks);
+        properties.put("mode", storageConfiguration.getFilterMode().toString());
+        properties.put("access_type", storageConfiguration.getAccessMode().toString());
+        properties.put("position", LuaConverter.posToObject(nodeContainer.getLocalPosition()));
+        properties.put("priority", nodeContainer.getPriority());
+
+        return properties;
+    }
+
+    public static Object parseStorageDisk(StateTrackedStorage disk) {
+        Map<String, Object> properties = new HashMap<>();
+
+        properties.put("used", disk.getStored());
+        properties.put("capacity", disk.getCapacity());
+        properties.put("state", disk.getState().toString());
+        if (disk.getDelegate() instanceof SerializableStorage serializableStorage) {
+            String type;
+            if (serializableStorage.getType() == StorageTypes.ITEM) {
+                type = "item";
+            } else if (serializableStorage.getType() == StorageTypes.FLUID) {
+                type = "fluid";
+            } else if (serializableStorage.getType() == ChemicalResourceType.STORAGE_TYPE) {
```

Comment:
again does this works without mekanism?

Reply:
It at least did in my testing
But I am gonna test this specific thing again